uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/UiAutomatorUtils.java

```diff
@@ -321,6 +321,22 @@ public static void handleButtonClickForObjectWithText(@NonNull final String text
         }
     }
 
+    /**
+     * Clicks the button element associated to the supplied text.
+     * Do not throw an exception if the button is not found.
+     *
+     * @param text the text on the button to click
+     */
+    public static void handleButtonClickForObjectWithTextSafely(@NonNull final String text) {
+        final UiObject button = obtainUiObjectWithText(text);
+
+        try {
+            button.click();
+        } catch (final UiObjectNotFoundException e) {
+            Logger.i(TAG, "Button with text \"" + text + "\" was not found: " + e.getMessage());
```

Comment:
Why do we need this safe functionality? I see you are using this during launching of edge app. Should we not simply fail if the "No, Thanks" button is not found?

Reply:
From my perspective, I don't think so. I say that because our test is not concerned with these popups that edge shows the user, we should not assert them as they do not interact with broker and they change depending on updates from Edge (they can't truely be predicted from our end, we basically change the check depending on whether or not they appear). I think it's fine to skip asserting some ui aspects of these tests if they are not tied to anything in our library or the overall token flows.

Reply:
the log level here should probably be at least `warn`

Reply:
Sure, let's do that.